TTHAnalysis/python/plotter/makeShapeCards.py

```diff
@@ -124,19 +126,19 @@ def rebin2Dto1D(h,funcstring):
     newh.SetLineColor(h.GetLineColor())
     return newh
 
-myout = outdir+"/common/" if len(masses) > 1 else outdir;
+myout = os.path.join(options.outdir,"common") if len(masses) > 1 else options.outdir;
 
 report={}
 if options.infile!=None:
-    infile = ROOT.TFile(myout+binname+".bare.root","read")
+    infile = ROOT.TFile(os.path.join(myout,binname+".bare.root"),"read")
     for p in mca.listSignals(True)+mca.listBackgrounds(True)+['data']:
         h = infile.Get(p)
         if h: report[p] = h
 else:
     report = mca.getPlotsRaw("x", args[2], args[3], cuts.allCuts(), nodata=options.asimov)
 
 if options.savefile!=None:
-    savefile = ROOT.TFile(myout+binname+".bare.root","recreate")
+    savefile = ROOT.TFile(os.path.join(myout,binname+".bare.root"),"recreate")
     for n,h in report.iteritems(): savefile.WriteTObject(h,n)
```

Comment:
`os.path.join(a,b)` just does `a+"/"+b`, at least on all operating systems we support, so I really doubt you need to change the code in 10 different places just to add one mkdir.